### PR TITLE
Add the validation for ResourceExploringWebhookConfiguration

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
Signed-off-by: changzhen <changzhen5@huawei.com>

**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Add the validation for `ResourceExploringWebhookConfiguration` creation/update, the restriction complies with the API definition.

For example, when I create a `ResourceExploringWebhookConfiguration` object with following yaml:
```yaml
// example-rewc.yaml
apiVersion: config.karmada.io/v1alpha1
kind: ResourceExploringWebhookConfiguration
metadata:
  name: example
webhooks:
  - name: foo.example.com
    rules:
      - operations: ["ExploreRetaining", "ExploreHealthy", "ExploreTest"]
        apiGroups: ["foo.example.com"]
        apiVersions: ["*"]
        resources: ["foos"]
        scope: "Namespaced"
    clientConfig:
      url: https://xxx:443/explore-foo
      caBundle: xxx
    failurePolicy: Fail
    exploreReviewVersions: ["v1alpha1"]
    timeoutSeconds: 3
```
Run `kubectl` to create a command prompt:
```
# kubectl create -f example-rewc.yaml 
Error from server (webhooks[0].rules[0].operations[2]: Unsupported value: "ExploreTest": supported values: "*", "ExploreDependencies", "ExploreHealthy", "ExplorePacking", "ExploreReplica", "ExploreReplicaRevising", "ExploreRetaining", "ExploreStatus", "ExploreStatusAggregating"): error when creating "example-rewc.yaml": admission webhook "config.karmada.io" denied the request: webhooks[0].rules[0].operations[2]: Unsupported value: "ExploreTest": supported values: "*", "ExploreDependencies", "ExploreHealthy", "ExplorePacking", "ExploreReplica", "ExploreReplicaRevising", "ExploreRetaining", "ExploreStatus", "ExploreStatusAggregating"
```

**Which issue(s) this PR fixes**:
Part of #923 

**Special notes for your reviewer**:

Some codes are copy from `https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/admissionregistration/validation/validation.go`, it has been explained.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by jrkeen — COMMENTED** [review]
lgtm

File: pkg/webhook/config/helper.go [hunk 1]
```diff
@@ -0,0 +1,257 @@
+/*
+   Copyright 2017 The Kubernetes Authors.
```
Comment (gy95) [inline pkg/webhook/config/helper.go:2]: this file is not used?

File: pkg/webhook/config/validating.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	klog.V(2).Infof("Validating ResourceExploringWebhookConfiguration(%s) for request: %s", configuration.Name, req.Operation)
+
+	var allErrors field.ErrorList
+	hookNames := sets.NewString()
+	for i, hook := range configuration.Webhooks {
+		allErrors = append(allErrors, validateWebhook(&configuration.Webhooks[i], field.NewPath("webhooks").Index(i))...)
+		if hookNames.Has(hook.Name) {
+			allErrors = append(allErrors, field.Duplicate(field.NewPath("webhooks").Index(i).Child("name"), hook.Name))
```
[the remark this answers is not in the document]
Comment (XiShanYongYe-Chang) [reply pkg/webhook/config/validating.go:40]: You are right 👍 .

File: pkg/webhook/config/helper.go (hunk 1 above)
Comment (XiShanYongYe-Chang) [reply pkg/webhook/config/helper.go:2]: It is used by `pkg/webhook/config/validating.go`.

Comment (gy95) [reply pkg/webhook/config/helper.go:2]: oh, i see.

**Pull request merged**